syntax = "proto3";
package prediction;

service ModelsService {
  rpc GetModelPrediction(ModelPredictionRequest)
      returns (ModelPredictionResponse);
  rpc GetModelPredictions(stream ModelPredictionRequest)
      returns (stream ModelPredictionResponse);
}

message Model { string uuid = 1; }

message ModelPredictionRequest {
  TimeSeriesValues time_series_values = 1;
  Model model = 2;
}

message TimeSeriesValues { repeated double values = 1; }

message ModelPredictionResponse { string classified_as = 1; }